core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -154,7 +168,7 @@ public ActionRequestValidationException validate() {
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         if (in.readBoolean()) {
-            internalShardId = ShardId.readShardId(in);
+            shardId = ShardId.readShardId(in);
         }
```

Comment:
fixed in https://github.com/areek/elasticsearch/commit/d6df74baebd52d26e0200d818e0e65fec442a749